server/services/import/import.js

```diff
@@ -56,7 +57,7 @@ const importData = async (dataRaw, { slug, format, user, idField }) => {
  * @param {string} idField - Field used as unique identifier.
  * @returns Updated/created entry.
  */
-const updateOrCreate = async (user, slug, data, idField = 'id') => {
+const updateOrCreate = async (user, slug, data, idField = getConfig('importUniqueIdentifierField')) => {
```

Comment:
The parameter should be used above in the function `importData`:

```js
await updateOrCreate(user, slug, datum, idField || getConfig('importUniqueIdentifierField'));
```

Since `updateOrCreate` is recursive and applied to relations as well, `importUniqueIdentifierField` will introduce breaking changes if used here

Reply:
Why? `importUniqueIdentifierField` has a default value `id`, so the plugin works as before without extra configuration.

BTW, I thought of a hierarchy of identifier fields, like id, then slug, then title or something like that. In my case it could help me create relations to all related data, even with entities without slugs.

Reply:
Currently, when you specify an `idField`, it only applies to the root content. Since relations may not have this `idField`, it falls back to `id` (`updateOrCreate` is recursive, see lines 131, 144 and 170). By modifying the default value of `idField` in `updateOrCreate`, `idField` will also be applied to relations which introduces an incorrect behavior.

That's why the change should be made on the function `importData` so it's only applied to the root content

Reply:
As for the hierarchy of identifiers, feel free to open an issue describing your needs in details 👍